src/api/test/unit/code_quality_test.rb

```diff
@@ -87,7 +87,7 @@ def setup
       'PersonController#internal_register'                                      => 112.01,
       'Package#find_changed_issues'                                             => 93.74,
       'Package#close_requests'                                                  => 84.82,
-      'Flag#default_status'                                                     => 90.56,
+      'Flag#compute_status'                                                     => 145.55,
```

Comment:
I know it's complicated. I also know the reasons. This was just an expression about my feelings for a complexity of 150 :-)